feat: introduce auth configuration

wolf4ood opened the pull request:
## What this PR changes/adds

Introduces a configuration extension for associating an auth implementation to a web context with configuration.

Introduces an `ApiAuthenticationProvider` which creates a `AuthenticationService` based on the input config.

The configuration plugs into the `web.http.<context>` mechanism adding a nested key for configuring the authentication for the specific `<context>`. For example for configuring the tokenbased auth for a custom context `custom`:

```
web.http.custom.path=/custom
web.http.custom.port=8081
web.http.custom.auth.type=tokenbased
web.http.custom.auth.key=apiKey
```

## Why it does that

flexibility, cleanup, decoupling

## Further notes

Some API configuration module explicitly register the `AuthenticationRequestFilter` and uses custom `context` name to be used in the `ApiAuthenticationRegistry`. For backward compatibility that has not been changed. For overriding the AuthenticationService for that specific api context the additional config value `context` is available

For example for the management context:

```
web.http.management.auth.context=management-api
```

As future improvement we should remove `ApiAuthenticationProvider` and apply directly the `AuthenticationRequestFilter` with the configured `AuthenticationService`. We also might want to incorporate
this extension as default behavior in the web extension.

## Linked Issue(s)

Closes #4294 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

ndr-brt reviewed (commented):
Great stuff!
I think it would be beneficial to have a piece of `.md` in the documentation that explains how to secure the APIs, it would be also good to mark the explicit registration of the filter as deprecated so we could get rid of them at the right time.

wolf4ood commented:
@ndr-brt 

added some docs on the md file

@paullatzelsperger 
I've added in the further notes the next step of refactoring that we could to to remove the `AuthRegistry`
